src/app/components/Core/LinkApp.tsx

```diff
@@ -0,0 +1,206 @@
+import React from 'react';
+
+import { Link as RouterLink, LinkProps as RouterLinkProps } from 'react-router-dom';
+import styled from 'styled-components';
+import {
+  color,
+  ColorProps,
+  colorStyle,
+  ColorStyleProps,
+  fontStyle,
+  FontStyleProps,
+  layout,
+  LayoutProps,
+  lineHeight,
+  LineHeightProps,
+  size,
+  SizeProps,
+  space,
+  SpaceProps,
+  textStyle,
+  TextStyleProps,
+  typography,
+  TypographyProps,
+  variant
+} from 'styled-system';
+
+type LinkStyleProps = SpaceProps &
+  LineHeightProps &
+  FontStyleProps &
+  SizeProps &
+  ColorProps &
+  ColorStyleProps &
+  TextStyleProps &
+  LayoutProps &
+  TypographyProps & {
+    variant?: keyof typeof LINK_VARIANTS;
+    $underline?: boolean;
+    $textTransform?: 'uppercase' | 'capitalize' | 'lowercase';
+  };
+
+const LINK_RECIPES = {
+  default: {
+    cursor: 'pointer',
+    transition: 'all 120ms ease',
+    textDecoration: 'none',
+    // https://mayashavin.com/articles/svg-icons-currentcolor
+    svg: {
+      fill: 'currentColor'
+    },
+    '&:hover svg': {
+      fill: 'currentColor'
+    }
+  }
+};
+
+const LINK_VARIANTS = {
+  barren: {
+    ...LINK_RECIPES.default,
+    color: 'inherit'
+  },
+  underlineLink: {
+    ...LINK_RECIPES.default,
+    color: 'inherit',
+    textDecoration: 'underline',
+    '&:hover': {
+      textDecoration: 'none'
+    }
+  },
+  opacityLink: {
+    ...LINK_RECIPES.default,
+    color: 'BLUE_SKY',
+    '&:hover': {
+      opacity: '0.8'
+    },
+    '&:hover svg': {
+      opacity: '0.8'
+    }
+  },
+  defaultLink: {
+    ...LINK_RECIPES.default,
+    fontSize: { _: 0, sm: 1 },
+    lineHeight: { _: 0, sm: 1 },
+    color: 'BLUE_BRIGHT',
+    '&:hover': {
+      color: 'BLUE_LIGHT_DARKISH'
+    },
+    '&:active': {
+      opacity: 1
+    },
+    '&:focus': {
+      opacity: 1
+    }
+  }
+};
+
+const SLink = styled('a')<LinkStyleProps & React.AnchorHTMLAttributes<HTMLAnchorElement>>`
```

Comment:
We should convert to the Rebass `Link` and get rid of the style-component/styled-system method